test/e2e/conformance_test.go

```diff
@@ -118,6 +118,15 @@ var _ = Describe("Conformance Tests", func() {
 			}
 		}
 
+		// use the ipv6 flavor if ipv6 IP family is specified.
+		if e2eConfig.GetVariable(capi_e2e.IPFamily) == "IPv6" {
+			flavor += "-ipv6"
+			kubetestConfigFilePath = strings.Replace(kubetestConfigFilePath, ".yaml", "-ipv6.yaml", 1)
```

Comment:
Quick clarifying question: Will this still find the right flavor if we use CI or PR artifacts? I don't see pre-existing templates for them even without the ip suffix so I'm wondering if this may cause problems if we're getting the templates from elsewhere.

The flavors being set: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/0151252433f9b0d284312fe0dc761173c4980d4a/test/e2e/conformance_test.go#L115

Reply:
Yes for CI, that's what this PR is enabling, the flavors are defined here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4086/files#diff-35f348a5d567ebeba39b1e40bbab0e1a7f64009cbadff142952f9b9691e94666R121

For PR artifacts no since those templates haven't been added. I could change the line so that it only changes the suffix for CI artifacts but then if you set "ipv6" and "pr artifacts" the test would silently fail (it would just use the regular template) instead of failing with a flavor not found error so I think it's arguably worse. This also is more future proof if we do add the equivalent ipv6/dual stack PR artifacts templates later.